src/app/clusters/temperature-measurement-server/temperature-measurement-server.cpp

```diff
@@ -15,78 +15,88 @@
 
 #include <app/util/af.h>
 
-#include <app/util/af-event.h>
-#include <app/util/attribute-storage.h>
-
+#include "gen/att-storage.h"
 #include "gen/attribute-id.h"
 #include "gen/attribute-type.h"
 #include "gen/cluster-id.h"
+#include "gen/command-id.h"
 
-using namespace chip;
+#include <support/logging/CHIPLogging.h>
 
-EmberEventControl emberAfPluginTemperatureMeasurementServerReadEventControl;
+#ifndef emberAfTemperatureMeasurementClusterPrintln
+#define emberAfTemperatureMeasurementClusterPrintln(...) ChipLogProgress(Zcl, __VA_ARGS__);
+#endif
 
-// TODO: There's no header that declares this event handler, and it's not 100%
-// clear where best to declare it.
-// https://github.com/project-chip/connectedhomeip/issues/3619
-void emberAfPluginTemperatureMeasurementServerReadEventHandler() {}
+EmberAfStatus emberAfTemperatureMeasurementClusterGetMeasuredValue(chip::EndpointId endpoint, int16_t * measuredValue)
+{
+    return emberAfReadServerAttribute(endpoint, ZCL_TEMP_MEASUREMENT_CLUSTER_ID, ZCL_TEMP_MEASURED_VALUE_ATTRIBUTE_ID,
+                                      (uint8_t *) measuredValue, sizeof(*measuredValue));
+}
 
-void emberAfPluginTemperatureMeasurementServerStackStatusCallback(EmberStatus status) {}
+EmberAfStatus emberAfTemperatureMeasurementClusterGetMinMeasuredValue(chip::EndpointId endpoint, int16_t * minMeasuredValue)
+{
+    return emberAfReadServerAttribute(endpoint, ZCL_TEMP_MEASUREMENT_CLUSTER_ID, ZCL_TEMP_MIN_MEASURED_VALUE_ATTRIBUTE_ID,
+                                      (uint8_t *) minMeasuredValue, sizeof(*minMeasuredValue));
+}
 
-// -------------------------------------------------------------------------
-// ****** callback section *******
+EmberAfStatus emberAfTemperatureMeasurementClusterGetMaxMeasuredValue(chip::EndpointId endpoint, int16_t * maxMeasuredValue)
+{
+    return emberAfReadServerAttribute(endpoint, ZCL_TEMP_MEASUREMENT_CLUSTER_ID, ZCL_TEMP_MAX_MEASURED_VALUE_ATTRIBUTE_ID,
+                                      (uint8_t *) maxMeasuredValue, sizeof(*maxMeasuredValue));
+}
 
-void emberAfPluginTemperatureMeasurementServerInitCallback(void)
+void emberAfTemperatureMeasurementClusterServerInitCallback(chip::EndpointId endpoint)
 {
-    EmberAfStatus status;
-    // FIXME Use real values for the temperature sensor polling the sensor using the
-    //       EMBER_AF_PLUGIN_TEMPERATURE_MEASUREMENT_SERVER_MAX_MEASUREMENT_FREQUENCY_S macro
-    EndpointId endpointId = 1; // Hardcoded to 1 for now
-    int16_t newValue      = 0x1234;
+    int16_t value        = 0;
+    EmberAfStatus status = emberAfTemperatureMeasurementClusterGetMeasuredValue(endpoint, &value);
+    if (EMBER_ZCL_STATUS_SUCCESS != status)
+    {
+        status = emberAfTemperatureMeasurementClusterSetMeasuredValueCallback(endpoint, INT16_MIN);
+        if (EMBER_ZCL_STATUS_SUCCESS != status)
+        {
+            emberAfTemperatureMeasurementClusterPrintln("ERR: writing present value %x", status);
```

Comment:
nit: Use a bit more meaningful message like "writing current temp value"? Similarly, below you could use "writing min temp value" and "writing max temp value".